lib/tasks/ci/licenses.rake

```diff
@@ -0,0 +1,23 @@
+namespace :ci do
+  namespace :license_finder do
+    desc 'Run compliance task and generates the license report if complies'
+    task :run do
+      if Rake::Task['ci:license_finder:compliance'].invoke
+        Rake::Task['ci:license_finder:report'].invoke
+      end
+    end
+    desc 'Check license compliance of dependencies'
+    task :compliance do
+      STDOUT.puts 'Checking license compliance'
+      unless system("bundle exec bin/license_finder")
+        STDERR.puts "*** License compliance test failed  ***"
+        exit 1
+      end
+    end
+    desc 'Generates a report with the dependencies and their licenses'
+    task :report do
+      STDOUT.puts 'Generating report...'
+      exec("bundle exec bin/license_finder report --format=xml > #{Rails.root.join('doc/licenses/licenses.xml')}")
```

Comment:
This can be done by the RUBYOPT instead of having a custom binary, right @ddcesare ?

Then the whole rake and binary could be gone.

It is not pretty, but works. So I'm ok with it, but not happy :)